newroutes/auth.js

```diff
@@ -0,0 +1,99 @@
+const autoBind = require("auto-bind")
+const express = require("express")
+
+// Import middleware
+const { attachReadRouteHandlerWrapper } = require("@middleware/routeHandler")
+
+const { FRONTEND_URL } = process.env
+
+const AUTH_TOKEN_EXPIRY_MS = parseInt(
+  process.env.AUTH_TOKEN_EXPIRY_DURATION_IN_MILLISECONDS,
+  10
+)
+const CSRF_TOKEN_EXPIRY_MS = 600000
+const CSRF_COOKIE_NAME = "isomer-csrf"
+const COOKIE_NAME = "isomercms"
+
+class AuthRouter {
+  constructor({ authService }) {
+    this.authService = authService
+    // We need to bind all methods because we don't invoke them from the class directly
+    autoBind(this)
+  }
+
+  async authRedirect(req, res) {
+    const {
+      redirectUrl,
+      cookieToken,
+    } = await this.authService.getAuthRedirectDetails()
+    const csrfTokenExpiry = new Date()
+    csrfTokenExpiry.setTime(csrfTokenExpiry.getTime() + CSRF_TOKEN_EXPIRY_MS)
+    const cookieSettings = {
+      expires: csrfTokenExpiry,
+      httpOnly: true,
+      secure:
+        process.env.NODE_ENV !== "DEV" &&
+        process.env.NODE_ENV !== "LOCAL_DEV" &&
+        process.env.NODE_ENV !== "test",
+    }
+    res.cookie(CSRF_COOKIE_NAME, cookieToken, cookieSettings)
+    return res.redirect(redirectUrl)
+  }
+
+  async githubAuth(req, res) {
+    const csrfState = req.cookies[CSRF_COOKIE_NAME]
+    const { code, state } = req.query
+
+    const token = await this.authService.getGithubAuthToken({
+      csrfState,
+      code,
+      state,
+    })
+    const authTokenExpiry = new Date()
+    authTokenExpiry.setTime(authTokenExpiry.getTime() + AUTH_TOKEN_EXPIRY_MS)
+    const cookieSettings = {
+      path: "/",
+      expires: authTokenExpiry,
+      httpOnly: true,
+      sameSite: true,
+      secure:
+        process.env.NODE_ENV !== "DEV" &&
+        process.env.NODE_ENV !== "LOCAL_DEV" &&
+        process.env.NODE_ENV !== "test",
```

Comment:
see above wrt helper function

Reply:
https://github.com/isomerpages/isomercms-backend/pull/328/commits/370e105d57b130be7ecda03850e5280a91fc4c70

Reply:
wdyt about shifting the helper to utils as it might be used elsewhere in the future? eslint also screams at me about `Expected 'this' to be used by class method 'isSecure'.`, which seems to hint that this shouldn't belong within the class